crypto/crypto/src/hash.rs

```diff
@@ -66,26 +66,15 @@
 //! **Note**: The last argument for the `define_hasher` macro must be a unique string.
 //!
 //! ## The `CryptoHash` implementation (for both automatic and semi-automatic way)
-//! Then, the `CryptoHash` trait should be implemented:
+//! In most cases, the `CryptoHash` trait should be implemented using LCS serialization.
+//! The derive macro `LCSCryptoHash` generates such an implementation based on `serde::Serialize`,
+//! the LCS encoding, and `MyNewStructHasher`:
 //! ```
 //! # use libra_crypto::hash::*;
-//! # #[derive(Default)]
-//! # struct MyNewStructHasher;
-//! # impl CryptoHasher for MyNewStructHasher {
-//! #   fn finish(self) -> HashValue { unimplemented!() }
-//! #   fn write(&mut self, bytes: &[u8]) -> &mut Self { unimplemented!() }
-//! # }
-//! struct MyNewStruct;
-//!
-//! impl CryptoHash for MyNewStruct {
-//!     type Hasher = MyNewStructHasher; // use the above defined hasher here
-//!
-//!     fn hash(&self) -> HashValue {
-//!         let mut state = Self::Hasher::default();
-//!         state.write(b"Struct serialized into bytes here");
-//!         state.finish()
-//!     }
-//! }
+//! # use libra_crypto_derive::*;
+//! # use serde::Serialize;
+//! #[derive(Serialize, CryptoHasher, LCSCryptoHash)]
+//! struct MyNewStruct { /* ... */ }
```

Comment:
That's great, but can we sketch the use of the hash?

Reply:
Good idea, thanks!